ASWF/CONTRIBUTING.md

```diff
@@ -0,0 +1,333 @@
+instructions on coding style below).
+
+1. Push your changes to your fork (each unrelated pull request to a separate
+"topic branch", please).
+
+1. Make a "pull request" on GitHub for your patch.
+
+2. If your patch will induce a major compatibility break, or has a design
+component that deserves extended discussion or debate among the wider OIIO
+community, then it may be prudent to email oiio-dev pointing everybody to
+the pull request URL and discussing any issues you think are important.
+
+1. All pull requests automatically launch CI jobs on GitHub Actions to
+ensure that the build completes and that the tests suite runs correctly, for
+a variety of platform, compiler, library, and flag combinations. The status
+of the CI tests for your PR will be displayed on the GitHub PR page. We will
+not accept PRs that don't build cleanly or pass the existing testsuite.
+
+1. The reviewer will look over the code and critique on the "comments" area.
+Reviewers may ask for changes, explain problems they found, congratulate the
+author on a clever solution, etc. But until somebody says "LGTM" (looks good
+to me), the code should not be committed. Sometimes this takes a few rounds
+of give and take. Please don't take it hard if your first try is not
+accepted. It happens to all of us.
+
+1. After approval, one of the senior developers (with commit approval to the
+official main repository) will merge your fixes into the master branch.
+
+
+Coding Style
+------------
+
+### File conventions
+
+C++ implementation should be named `*.cpp`. Headers should be named `.h`.
+All headers should contain
+
+    #pragma once
+
+All source files should begin with these three lines:
+
+    // Copyright Contributors to the OpenImageIO project.
+    // SPDX-License-Identifier: BSD-3-Clause
+    // https://github.com/OpenImageIO/oiio
+
+as a comment in the syntax of whatever source code is used in that file.
+
+Occasionally a file may contain substantial code from another project and will
+also list its original copyright and license information. Do NOT alter that
+notice or copy it to any new files, it really only applies to the particular
+file in which it appears.
+
+
+### Formatting
+
+We use [clang-format](https://clang.llvm.org/docs/ClangFormat.html)
+to uniformly format our source code prior to PR submission. Make sure that
+clang-format is installed on your local machine, and just run
+
+    make clang-format
+
+and it will automatically reformat your code according to the configuration
+file found in the `.clang-format` file at the root directory of the OIIO
+source code checkout.
+
+One of the CI test matrix entries runs clang-format and fails if any
+diffs were generated (that is, if any of your code did not 100% conform to
+the `.clang-format` formatting configuration). If it fails, clicking on that
+test log will show you the diffs generated, so that you can easily correct
+it on your end and update the PR with the formatting fixes.
```

Comment:
Different versions of clang-format can produce different output. Is it worth pointing this out in case folks hit this and don't understand why it's failing?

Reply:
I'll edit this to spell it out, but you are correct -- at any given time, there is a *specific* version of clang-format that is the ultimate source of truth in case other clang-format versions contradict each other, and it is the version we use for the "clang-format" CI test. I tend to keep the version steady for long periods, and should only change by PR in which it is deliberately bumping the clang-format version being used and discussing the implications of that (and usually that would require checking in an assortment of other changes across the codebase for the small number of things that might change their formatting).